boundary.go

```diff
@@ -181,10 +181,20 @@ func (b *boundaryReader) Next() (bool, error) {
 		_, _ = io.Copy(ioutil.Discard, b)
 	}
 	for {
-		line, err := b.r.ReadSlice('\n')
-		if err != nil && err != io.EOF {
-			return false, errors.WithStack(err)
+		var line []byte
+		var err error
+		for {
+			// Read whole line, handle extra long lines in cycle
+			var segment []byte
+			segment, err = b.r.ReadSlice('\n')
+			line = append(line, segment...)
```

Comment:
A concern I have here is we are going to be doing a lot more allocations than we used to.  In the previous version of this loop, `line` was just a pointer into the bufio buffer, but now we are allocating and growing `line` multiple times per Next().

Is it feasible to move the line buffer into a `b` struct and re-use it?

Reply:
What about allocating only when the line is longer than buffer?

In fact, this is not a bug at all. MIME mail must have at most 998 octects on the line according to RFC, so it's about tolerance of some existing malformed emails.